azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,406 @@
+		server, err := client.GetDetails(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Analysis Services Server %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_analysis_services_server", *server.ID)
+		}
+	}
+
+	sku := d.Get("sku").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+
+	serverProperties := expandAnalysisServicesServerProperties(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	analysisServicesServer := analysisservices.Server{
+		Name:             &name,
+		Location:         &location,
+		Sku:              &analysisservices.ResourceSku{Name: &sku},
+		ServerProperties: serverProperties,
+		Tags:             expandTags(tags),
+	}
+
+	future, err := client.Create(ctx, resourceGroup, name, analysisServicesServer)
+	if err != nil {
+		return fmt.Errorf("Error creating Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	server, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if server.ID == nil {
+		return fmt.Errorf("Cannot read ID of Analysis Services Server %q (Resource Group %q)", name, resourceGroup)
+	}
+
+	d.SetId(*server.ID)
+
+	return resourceArmAnalysisServicesServerRead(d, meta)
+}
+
+func resourceArmAnalysisServicesServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	server, err := client.GetDetails(ctx, resourceGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(server.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure Analysis Services Server %q: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := server.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	d.Set("sku", *server.Sku.Name)
+
+	if serverProps := server.ServerProperties; serverProps != nil {
+		if serverProps.AsAdministrators == nil || serverProps.AsAdministrators.Members == nil {
+			d.Set("admin_users", []string{})
+		} else {
+			d.Set("admin_users", *serverProps.AsAdministrators.Members)
+		}
+
+		enablePowerBi, fwRules := flattenAnalysisServicesServerFirewallSettings(serverProps)
+		d.Set("enable_power_bi_service", enablePowerBi)
+		d.Set("ipv4_firewall_rule", fwRules)
```

Comment:
since this is a complex object could we check for errors when setting here?

```suggestion
		if err := d.Set("ipv4_firewall_rule", fwRules); err != nil {
			return fmt.Errorf("Error setting `ipv4_firewall_rule`: %s", err)
		}
```